src/pl/senla/hotel/ie/file/EntitiesConverter.java

```diff
@@ -12,74 +13,34 @@
 import pl.senla.hotel.utils.RoomReservationUtil;
 import pl.senla.hotel.utils.RoomUtil;
 
-import java.io.BufferedReader;
-import java.io.FileReader;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-
 import static pl.senla.hotel.constant.InputOutputConstant.*;
 
-public class ReaderWriter<T> implements ReaderWriterUniversal<T>{
+public class EntitiesConverter<T> implements Converter<T> {
```

Comment:
I'm sorry, Mark. probably I didn't catch your comment.
My ReaderWriter looked so:
public class ReaderWriter<T> implements ReaderWriterUniversal{
    private final Converter<T> converter;
    public ReaderWriter() {
        this.converter = new EntitiesConverter<>();
    }

Should I add     "Converter<T> converter"    as a parametr to constructor?
public ReaderWriter(Converter<T> converter) {
        this.converter = new EntitiesConverter<>();
    }

Did I understand correctly?

Reply:
The idea is that the reader only reads your CSV file, transfers the content to the converter, and it converts it into the desired entity and returns it.
With this approach, we can easily comply with the SOLID principles and get rid of code duplication (or large switch/if blocks) using abstract classes.

Yes, you must pass a certain converter to the reader, for certain entities, this can be done through a constructor.

Reply:
ready.